cookbook/common/Makefile

```diff
@@ -0,0 +1,183 @@
+	echo "REGISTRY ${REGISTRY}"
+
+TAGGED_IMAGE=${FULL_IMAGE_NAME}:${PREFIX}-${VERSION}
+
+# This should only be used by Admins to push images to the public Dockerhub repo. Make sure you
+# specify REGISTRY=ghcr.io/flyteorg or your registry before the make command otherwise this won't actually push
+# Also if you want to push the docker image for sagemaker consumption then
+# specify ECR_REGISTRY
+.PHONY: docker_push
+docker_push: docker_build
+ifdef REGISTRY
+	docker push ${TAGGED_IMAGE}
+endif
+
+.PHONY: fmt
+fmt: # Format code with black and isort
+	black .
+	isort .
+
+.PHONY: install-piptools
+install-piptools:
+	pip install -U pip-tools
+
+.PHONY: setup
+setup: install-piptools # Install requirements
+	pip-sync dev-requirements.txt
+
+.PHONY: lint
+lint:  # Run linters
+	flake8 .
+
+requirements.txt: export CUSTOM_COMPILE_COMMAND := $(MAKE) requirements.txt
+requirements.txt: requirements.in install-piptools
+	$(call PIP_COMPILE,requirements.in)
+
+.PHONY: requirements
+requirements: requirements.txt
+
+.PHONY: fast_serialize
+fast_serialize: clean _pb_output
+	echo ${CURDIR}
+	docker run -it --rm \
+		-e MAKEFLAGS \
+		-v ${CURDIR}/_pb_output:/tmp/output \
+		-v ${CURDIR}:/root/$(shell basename $(CURDIR)) \
+		${TAGGED_IMAGE} make fast_serialize
+
+.PHONY: fast_register
+fast_register: clean _pb_output ## Packages code and registers without building docker images.
+	@echo "Tagged Image: "
+	@echo ${TAGGED_IMAGE}
+	@echo ${CURDIR}
+	docker run -it --rm \
+		--network host \
+		-e REGISTRY=${REGISTRY} \
+		-e MAKEFLAGS=${MAKEFLAGS} \
+		-e FLYTE_HOST=${FLYTE_HOST} \
+		-e INSECURE_FLAG=${INSECURE_FLAG} \
+		-e PROJECT=${PROJECT} \
+		-e FLYTE_AWS_ENDPOINT=${FLYTE_AWS_ENDPOINT} \
+		-e FLYTE_AWS_ACCESS_KEY_ID=${FLYTE_AWS_ACCESS_KEY_ID} \
+		-e FLYTE_AWS_SECRET_ACCESS_KEY=${FLYTE_AWS_SECRET_ACCESS_KEY} \
+		-e OUTPUT_DATA_PREFIX=${OUTPUT_DATA_PREFIX} \
+		-e ADDL_DISTRIBUTION_DIR=${ADDL_DISTRIBUTION_DIR} \
+		-e SERVICE_ACCOUNT=$(SERVICE_ACCOUNT) \
+		-v ${CURDIR}/_pb_output:/tmp/output \
+		-v ${CURDIR}:/root/$(shell basename $(CURDIR)) \
+		${TAGGED_IMAGE} make fast_register
+
+.PHONY: docker_build
+docker_build:
+	echo "Tagged Image: "
+	echo ${TAGGED_IMAGE}
+	docker build ../ --build-arg tag="${TAGGED_IMAGE}" -t "${TAGGED_IMAGE}" -f Dockerfile
+
+.PHONY: serialize
+serialize: clean _pb_output docker_build
+	@echo ${VERSION}
+	@echo ${CURDIR}
+	docker run -it --rm \
+		-e MAKEFLAGS \
+		-v ${CURDIR}/_pb_output:/tmp/output \
+		${TAGGED_IMAGE} make serialize
+
+
+.PHONY: register
+register: clean _pb_output docker_push
+	@echo ${VERSION}
+	@echo ${CURDIR}
+	docker run -it --rm \
+		--network host \
+		-e REGISTRY=${REGISTRY} \
+		-e MAKEFLAGS \
+		-e FLYTE_HOST \
+		-e INSECURE_FLAG \
+		-e PROJECT \
+		-e FLYTE_AWS_ENDPOINT \
+		-e FLYTE_AWS_ACCESS_KEY_ID \
+		-e FLYTE_AWS_SECRET_ACCESS_KEY \
+		-e OUTPUT_DATA_PREFIX \
+		-e ADDL_DISTRIBUTION_DIR \
+		-e SERVICE_ACCOUNT=$(SERVICE_ACCOUNT) \
+		-e VERSION=${VERSION} \
```

Comment:
Should we explicitly specify these environment variables like we have done for `fast_register`?